### PR TITLE
Checkout zone

### PR DESCRIPTION
**Pull request by RebeccaStankus**
Recycle zone and Scanner zone will only search for overlays while an avatar is in the checkout zone with this PR. This PR includes the updated scripts for the recycle zone and scanner zone, an updated checkout zone script that calls methods from recycle and scanner when an avatar enters the zone, and an updated JSON for the entire checkout area including mirror, checkout stand, and all relevant zones.  Recycle zone now deletes the item on the avatar when it's matching overlay is deleted. Tablet will redirect to wallet app if the user's wallet is not set up and authenticated. An item's position on the avatar will be stored upon scanning the item so that it can be rezzed in the same spot. This allows for adjustment inside the checkout zone.

TEST PLAN
-Have two users go to dev-content, A&J store.
-User one will need to open and close the tablet before proceeding to get accurate results.
-User one enters checkout having set up wallet in previous session, but not having authenticated during this session
-Upon entering checkout area, tablet will open to authentication page.
-Repeat with user one having set up wallet and authenticated during the current session.
-Upon entering checkout area, tablet will open to checkout instructions page.
-User one tries on a the monster beanie, cowboy hat, and purple scarf.
-User one goes to the checkout zone and stands in front of the middle checkout stand.
-User one should see repeating messages in Debug Window about the scanner and recycle bin searching for items.
-User two should not see these messages.
-User one grabs an overlay, places it in the recycle bin, and releases it.
-The overlay and its matching item on the avatar should both disappear.
-User one places an overlay in the scanner area.
-The tablet should redirect to the marketplace page of the item in scanner and the overlay in the scanner should disappear.
-User one purchases the item and then clicks "Wear it"
-Item should briefly disappear and then reappear in the same position it was in when its matching overlay was scanned.
-Upon User one leaving the checkout area, the messages about the scanner and recycle bin should stop.

### CODE REVIEW

**Comment by misslivirose** [comment]
@RebeccaStankus which checkout zone in dev-content should be used for testing?

**Comment by RebeccaStankus** [comment]
I updated the test plan to include that the user should test on the middle checkout stand.

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 1]
```diff
@@ -17,53 +17,182 @@
...
+    // var _THIS = this;
+    var SEARCH_RADIUS = 2;
+    var MAKING_SURE_INTERVAL = 100; // Milliseconds
+    var STOP_MAKING_SURE_TIMEOUT = 5000; // Milliseconds
 
     var interval;
-    var properties;
     var checkoutZoneID;
+    var properties;
+    // var checkoutZoneID;
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: please remove unused commented code

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 2]
```diff
@@ -17,53 +17,182 @@
...
         onEntityAdded: function(entityID) {
-            var newItemProperties = Entities.getEntityProperties(entityID, 
-                ['marketplaceID', 'clientOnly', 'owningAvatarID', 'lastEditedBy']);
-            if ((newItemProperties.lastEditedBy === MyAvatar.sessionUUID || newItemProperties.lastEditedBy === undefined)
-             && newItemProperties.marketplaceID !== "") {
+            print("there's a new entity on the scene!!!");
+            var newItemProperties = Entities.getEntityProperties(entityID, ['marketplaceID', 'clientOnly', 
+                'owningAvatarID', 'lastEditedBy']);
+            // if item was made by current avatar and has a marketplace ID
+            if (
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: This if statement is formatted strangely

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 3]
```diff
@@ -17,53 +17,182 @@
...
+        getTransformForMarketplaceItems: function() {
+            return Settings.getValue(TRANSFORMS_SETTINGS, {});
+        },
+
         onEntityAdded: function(entityID) {
-            var newItemProperties = Entities.getEntityProperties(entityID, 
-                ['marketplaceID', 'clientOnly', 'owningAvatarID', 'lastEditedBy']);
-            if ((newItemProperties.lastEditedBy === MyAvatar.sessionUUID || newItemProperties.lastEditedBy === undefined)
-             && newItemProperties.marketplaceID !== "") {
+            print("there's a new entity on the scene!!!");
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: remove print statements

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 4]
```diff
@@ -17,53 +17,182 @@
...
+            if (certificateID !== undefined) {
+                marketplaceItemTransform.certificateTransforms[certificateID] = transform;
+            } else {
+                marketplaceItemTransform.unsortedTransforms.push(transform);
+            }
+            marketplaceItemTransforms[marketplaceID] = marketplaceItemTransform;
+            Settings.setValue(TRANSFORMS_SETTINGS, marketplaceItemTransforms);
+        },
+
+        // TODO FIX DEMO ENTITY ID
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: Is this fixed?

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 5]
```diff
@@ -17,53 +17,182 @@
...
+            var makeSureInterval = Script.setInterval(function() {
+                Entities.editEntity(newEntityID, transformProperties);
+            }, MAKING_SURE_INTERVAL);
+    
+            // Five seconds should be enough to be sure, otherwise we have a problem
+            Script.setTimeout(function() {
+                makeSureInterval.stop();
+            }, STOP_MAKING_SURE_TIMEOUT);
+    
+            print("stored transforms again:  " + JSON.stringify(replicaStoredTransforms));
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: I don't think we need all of these print statements

File: DomainContent/AvatarStore/CheckoutZone/checkoutZone.js [hunk 6]
```diff
@@ -10,32 +10,33 @@
...
     var OVERLAY_ROTATIONAL_OFFSET = { x: 10, y: 140, z: 0 };
     var TABLET_LOCAL_POSITION_OFFSET = { x: 0.01, y: 0.9, z: -0.6 };
     var APP_ICON = "https://hifi-content.s3.amazonaws.com/rebecca/CheckoutZone/shoppingCart.svg";
     var TABLET = Tablet.getTablet("com.highfidelity.interface.tablet.system");
     var TABLET_ROTATIONAL_OFFSET = { x: 10, y: 240, z: 0 };
+    var MARKETPLACE_WALLET_QML_PATH = Script.resourcesPath() + "qml/hifi/commerce/wallet/Wallet.qml";
     // Milliseconds
     var MAKING_SURE_INTERVAL = 100;
-    var STOP_MAKING_SURE_TIMEOUT = 5000; 
+    // var STOP_MAKING_SURE_TIMEOUT = 5000;
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutZone.js]: remove unneeded commented code

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 7]
```diff
@@ -122,51 +108,39 @@
...
             Entities.deleteEntity(matchingEntity);
         },
 
         enterCheckout: function(entityID) {
             properties = Entities.getEntityProperties(entityID, ['position', 'parentID']);
             checkoutZoneID = properties.parentID;
             interval = Script.setInterval(function() {
-                // scanner is searching
-                print("Scanner is searching for overlays");
+                print("Scanner is searching");
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: Please remove this -- especially since it's on an interval.

**Comment by misslivirose** [comment]
@RebeccaStankus -- I tried this out and it opened up my wallet to my purchases page when I entered the checkout zone. When I left and re-entered it showed the marketplace landing page.

**Comment by misslivirose** [comment]
@RebeccaStankus -- how does this PR effect PR 46?

**Pull request closed without merging**